src/main/java/spoon/support/reflect/declaration/CtEnumImpl.java

```diff
@@ -113,8 +113,12 @@ public boolean removeEnumValue(CtEnumValue<?> enumValue) {
 
 	@Override
 	public <C extends CtEnum<T>> C setEnumValues(List<CtEnumValue<?>> enumValues) {
+		if (enumValues == null) {
```

Comment:
Why does this remove the static analysis warning?

Reply:
The same pattern here:
```
.... new ArrayList<>(enumValues) // <= if enumValues is null, we will get NullPointerException
if (enumValues == null || enumValues.isEmpty()) { .... } // <= enumValues could not be null here
```

We detect such errors with data-flow analysis. 
When I add that check for null with unconditional return in 'then' branch, the analyzer understands that in the code, following the check, the `enumValues` reference is not null.

Reply:
Thanks. I was not aware that `new ArrayList<>(null)` throws an NPE